src/main/java/Deadline.java

```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task {
+    private String name;
+    private String by;
```

Comment:
(extending this comment to the Event)
Might it be worth changing the date fields to things like `byDate`, `fromDate` etc? To more properly represent that these are indeed dates?